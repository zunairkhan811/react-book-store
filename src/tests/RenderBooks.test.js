import React from 'react';
import { useDispatch } from 'react-redux';
import { render } from '@testing-library/react';
import Render from '../components/RenderBooks';

jest.mock('react-redux', () => ({
  useDispatch: jest.fn(),
}));
jest.mock('../redux/books/removeDataFromApi', () => jest.fn());

describe('Render Component', () => {
  beforeEach(() => {
    useDispatch.mockReturnValue(jest.fn());
  });

  it('should render Render component', () => {
    const props = {
      id: '1',
      title: 'Book 1',
      author: 'Author 1',
      category: 'Category 1',
      key: '1',
    };

    const { container } = render(<Render {...props} />);
    expect(container).toMatchSnapshot();
  });
});