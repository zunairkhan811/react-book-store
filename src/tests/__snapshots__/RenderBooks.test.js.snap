// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Component should render Render component 1`] = `
<div>
  <div
    class="card-container"
    id="1"
  >
    <div
      class="info-div"
    >
      <div
        class="remove-div"
      >
        <h1>
          Category 1
        </h1>
        <h2>
          Book 1
        </h2>
        <p>
          Author 1
        </p>
        <button
          class="remove-btn"
          type="button"
        >
          Comments
        </button>
        <button
          class="remove-btn"
          type="button"
          value="Remove"
        >
          Remove
        </button>
        <button
          class="remove-btn"
          type="button"
        >
          Edit
        </button>
      </div>
      <div
        class="complete-div"
      >
        <div
          class="oval"
        />
        <span>
          <p>
            100%
          </p>
          <p>
            Completed
          </p>
        </span>
      </div>
    </div>
    <div
      class="update-div"
    >
      <h2>
        Current Chapter
      </h2>
      <h2>
        Chapter 17
      </h2>
      <button
        class="update-btn"
        type="button"
      >
        Update Progress
      </button>
    </div>
  </div>
</div>
`;
